mobile/library/common/engine_common.h

```diff
@@ -41,11 +41,11 @@ class EngineCommon {
   Envoy::SignalAction handle_sigs_;
   Envoy::TerminateHandler log_on_terminate_;
 #endif
-  std::unique_ptr<Envoy::OptionsImplBase> options_;
+  std::shared_ptr<Envoy::OptionsImplBase> options_;
```

Comment:
why did this need to be changed to shared_ptr?

Reply:
I have a comment about that in `internal_engine.cc`. This is because there's no way to do a move capture for `std::unique_ptr` with `std::function`. The `std::shared_ptr` is a workaround. I tried switching `std::function` to `absl::AnyInvocable` but ran into MSAN, ASAN, and TSAN. So, the safest way is to keep the existing behavior with `std::function`.

```
// This function takes a `std::shared_ptr` instead of `std::unique_ptr` because `std::function` is a
// copy-constructible type, so it's not possible to move capture `std::unique_ptr` with
// `std::function`.
```